Makefile

```diff
@@ -0,0 +1,163 @@
+BACKEND_SERVICES ?= autoignore autostarter configuration database detection fileobserver mitigation notifier prefixtree
+TAP_SERVICES ?= bgpstreamhisttap bgpstreamkafkatap bgpstreamlivetap exabgptap riperistap
+FRONTEND_SERVICES ?= frontend
+
+SERVICES ?= $(BACKEND_SERVICES) + $(TAP_SERVICES) + $(FRONTEND_SERVICES)
+
+PUSH ?= false
+BUILD_TAG ?= latest
+CONTAINER_REPO ?= docker.io/inspiregroup
+RELEASE ?= latest
+
+.PHONY: $(BACKEND_SERVICES)
+$(BACKEND_SERVICES): # build backend container
+$(BACKEND_SERVICES):
+	@echo "Building $@ service for tag $(BUILD_TAG)"
+	@docker pull $(CONTAINER_REPO)/artemis-$@:latest || true
+ifneq ($(BUILD_TAG), "latest")
+	@docker pull $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) || true
+endif
+	@docker build -t artemis-$@:$(BUILD_TAG) \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:latest \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) \
+		backend-services/$@/
+ifeq ($(PUSH), "true")
+	@docker tag artemis-$@:$(BUILD_TAG) $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+	@docker push $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+endif
+
+.PHONY: $(TAP_SERVICES)
+$(TAP_SERVICES): # build tap container
+$(TAP_SERVICES):
+	@echo "Building $@ service for tag $(BUILD_TAG)"
+	@docker pull $(CONTAINER_REPO)/artemis-$@:latest || true
+ifneq ($(BUILD_TAG), "latest")
+	@docker pull $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) || true
+endif
+	@docker build -t artemis-$@:$(BUILD_TAG) \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:latest \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) \
+		monitor-services/$@/
+ifeq ($(PUSH), "true")
+	@docker tag artemis-$@:$(BUILD_TAG) $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+	@docker push $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+endif
+
+.PHONY: $(FRONTEND_SERVICES)
+$(FRONTEND_SERVICES): # build frontend container
+$(FRONTEND_SERVICES):
+	@echo "Building $@ service for tag $(BUILD_TAG)"
+	@docker pull $(CONTAINER_REPO)/artemis-$@:latest || true
```

Comment:
When pulling from the official repo, we should use `tempfrontend`

Reply:
it doesn't matter as it will just skip using the latest image as cache.